jax/lax_reference.py

```diff
@@ -12,92 +12,175 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
```

Comment:
I wonder if we should rename this file? It's no longer a test-only implementation of lax primitives.

Reply:
Now I moved the `np_impl` rules out of it, it is again, so we can keep the name.